internal/util/strings.go

```diff
@@ -13,3 +13,25 @@ func CamelToSnake(str string) string {
 	snake = matchAllCap.ReplaceAllString(snake, "${1}_${2}")
 	return strings.ToLower(snake)
 }
+
+// JoinWithTruncation joins a slice of strings with a separator, truncating the middle if the slice is longer
+// than maxLength, using abbreviation as a placeholder for the truncated part. The last element of the slice is always
+// included in the result. For example: ["1", "2", "3", "4"] with sep=",", maxLength=3 and abbreviation="..." will
+// return "1, 2, ..., 4".
+func JoinWithTruncation(str []string, sep string, maxLength int, abbreviation string) string {
```

Comment:
The parameter names threw me off a bit here. I'd recommend:

* `strings` or `elements` instead of `str`; while reading the code I kept forgetting that `str` is a slice and not a string. 
* `maxStrings` or `maxElements` instead of `maxLength`; I initially misinterpreted maxLength as the maximum length of the string to output rather than the maximum number of elements that will be joined.